avaje-config/src/main/java/io/avaje/config/ConfigParsers.java

```diff
@@ -6,13 +6,13 @@
 import java.util.Set;
 
 /**
- * Holds the non-properties ConfigParsers.
+ * Holds the ConfigParsers for various extension types.
  */
-final class Parsers {
+public final class ConfigParsers {
```

Comment:
When we write a library like avaje-config we desire to not expose concrete classes to the public.

I suspect this concrete class should go back to being Parsers and not public and instead we add a new interface called ConfigParsers for it to implement [and the ConfigParsers interface is then exposed to implementations of URIConfigLoader].

You might suggest that adding an interface is overkill. I still probably prefer a public interface here.

Reply:
You might be surprised but I heavily agree. 

I don't like the idea of something switching from class -> interface later on. Although in practice it doesn't break compile time compat I still feel like there are unknowns.

Also let us say you switch to an Enum or Record. Now you have accidentally exposed methods or what you should not have.

Reply:
That being said I still probably have the interface `sealed` for now.

Reply:
> sealed

this is java 11